File: app/src/main/java/seigneur/gauvain/mycourt/data/repository/TempDataRepository.java
package seigneur.gauvain.mycourt.data.repository;

import javax.inject.Inject;
import javax.inject.Singleton;

import seigneur.gauvain.mycourt.data.model.Draft;
import seigneur.gauvain.mycourt.data.model.Shot;

/**
 * Class used to get reference of data which are not
 * saved in DB or SharedPrefs
 * eg. data send between presenters
 */
@Singleton
public class TempDataRepository {

    //Used to temporarily store Shot in order to send it between presenter...
    public Shot shot;
    //Used to temporarily store ShotDraft in order to send it between presenter...
    public Draft mShotDraft;

    //Set to define the source of the edition to register the Draft in the right way
    public int mDraftCallingSource;

    //Used to notify that change has been made on Draft DB
    //public boolean mDraftsChanged=false;

    @Inject
    public TempDataRepository() {}

    public void setShot(Shot inShot){
        shot=inShot;
    }

    public Shot getShot() {return shot;}

    public void setShotDraft(Draft shotDraft){
        mShotDraft=shotDraft;
    }

    public Draft getShotDraft() {return mShotDraft;}

    public void setDraftCallingSource(int draftCallingSource) {
        mDraftCallingSource= draftCallingSource;
    }

    public int getDraftCallingSource() {
        return mDraftCallingSource;
    }

}

File: app/src/main/java/seigneur/gauvain/mycourt/data/repository/TempDataRepository.kt
package seigneur.gauvain.mycourt.data.repository

import androidx.lifecycle.MutableLiveData
import javax.inject.Inject
import javax.inject.Singleton

import seigneur.gauvain.mycourt.data.model.Draft
import seigneur.gauvain.mycourt.data.model.Shot
import seigneur.gauvain.mycourt.utils.SingleLiveEvent

/**
 * Class used to get reference of data which are not
 * saved in DB or SharedPrefs
 * eg. data send between viewModel
 */
@Singleton
class TempDataRepository

@Inject
constructor() {

    //Used to temporarily store Shot in order to send it between viewModel
    lateinit var shot: Shot

    //Used to temporarily store ShotDraft in order to send it between viewModel
    lateinit var shotDraft: Draft

    //Set to define the source of the edition to register the Draft in the right way
    var draftCallingSource: Int = 0

    //Notify subscribers Tat action mode has been triggered or stopped
    var editMode = MutableLiveData<Int>()

    //delete selected item from draft fragment
    var deleteSelectedListCmd = SingleLiveEvent<Void>()

}
